File: Scripts/laptop/inheritance-multiple/javascript/inheritance-multiple.js
class Device {
	print() {
		console.log("Just a random device") 
	}
}


//base class
const MobilePhone = (MobilePhone) => class extends MobilePhone {

	call() {
		console.log("Pick up a phone");
	}

}

//base class
const Camera = (Camera) => class extends Camera {

	shootPictures() {
		console.log("Say cheese")
	}

}


//multiple inheritance
class CameraPhone extends MobilePhone(Camera(Device)) {

}

for (var i = 0; i < 100000000; ++i) {
const ziaomi = new CameraPhone(); 
ziaomi.shootPictures();
ziaomi.call();
}

File: Scripts/server/inheritance-multiple/swift/inheritance-multiple.swift
class Camera {
    func shootPicture() {
        print("Say cheese\n")
    }
}

protocol Phone {
    func call()
}

extension Phone {
    func call() {
        print("Pick a call\n")
    }
}

class CameraPhone: Camera, Phone {
   
   func doActivcity() {
        self.call()
        self.shootPicture()
   }
}


for i in 0...1000000 {
  let obj = CameraPhone()
  obj.doActivcity()
}